\chapter{Driver mutations} \label{chap:driver_mutations}

\section{Mutations}

\subsection{Cell signaling and signaling pathways}

\textbf{Cell signaling} is the process by which cells interact with each other, themselves, or their environment. This involves the transduction of signals, which can be chemical, or can involve other types such as pressure, temperature, or light signals \cite{cell_signaling}. \textbf{Pathways} are sequences of molecular interactions within a cell that lead to a change in the cell or the production of a specific product \cite{pathway}. These pathways have a direction in which the actions occur, with the terms \textit{upstream} and \textit{downstream} indicating the initial and final stages of these processes, respectively.

In cancer research, \textbf{signaling pathways} are of particular interest because they mediate the transduction of cell signals. Identifying and targeting the signaling pathways responsible for cancer growth could potentially halt the development of the disease. \todo{\href{https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8002322/}{check this out}, also check if what i wrote is actually true, i think i read it somewhere but can't find the source right now; expand on cell signaling? expand of pathways? if yes, make subsections}

\subsection{Passenger and driver mutations}

There are two types of mutations in cancer: \textbf{passenger mutations} and \textbf{driver mutations}. Passenger mutations do not confer direct benefits to tumor growth or development, whereas driver mutations actively contribute to cancer progression by providing an evolutionary advantage and promoting the proliferation of tumor cells. A \textbf{driver gene} is a gene that harbors at least one driver mutation, though it may also contain passenger mutations \todo{\href{https://www.aiom.it/wp-content/uploads/2019/02/20190524RM_21_Tommasi.pdf}{DO I ADD THIS AS A CITATION???}}. A driver pathway consists of at least one driver gene. Driver mutations, genes, and pathways are of significant scientific interest due to their crucial role in cancer proliferation.

Driver genes can be classified into 12 signaling pathways, which regulate cellular functions related to survival, fate, and genomic maintenance. \todo{use (and expand) this? same source as prev}

\section{Classifying mutations}

\subsection{Frequency}

To classify mutations into the two categories described, assessing their biological function is essential, though this remains a challenging task. Numerous methods exist to predict the functional impact of mutations based on \textit{a priori} knowledge. However, these approaches often fail to integrate information effectively across various mutation types and are limited by their reliance on known proteins, rendering them less effective for less-studied ones \cite{multi-dendrix}.

With the decreasing cost of DNA sequencing, it is now possible to categorize mutations by examining their frequency, as driver mutations are typically the most recurrent in patients' genomes \cite{multi-dendrix}. In fact, key driver events, such as TP53 loss-of-function mutations, can be identified by their significantly high frequency of occurrence across a set of tumors \cite{mutex}. However, in many cases, since driver mutations are predominantly located in genes that are part of cell signaling pathways, different patients may harbor mutations in different pathway loci. Indeed, driver mutations can vary extensively between patient samples, even within the same cancer type \cite{multi-dendrix}; additionally, there is minimal overlap of mutated genes across sample pairs, even from the same patient \cite{mdpfinder}, reducing the statistical power of frequency analyses.

Moreover, multiple alternative driver alterations in different genes may lead to similar downstream effects. In such instances, the selective advantage is distributed among the alterations frequences of these genes. In current cancer genomics studies, where the number of samples is significantly smaller than the number of genes profiled per sample, frequency-based methods lack the statistical efficacy to distinguish passenger and driver mutations \cite{mutex}. 

Therefore, studies should be conducted at the pathway level, as it is well established that different mutations can affect the same pathway across multiple samples \cite{multi-dendrix}. However, since each pathway involves multiple genes, numerous possible combinations of driver mutations could impact a crucial cancer pathway, making it computationally unfeasible to test every possible gene permutation \cite{dendrix} --- estimates suggest that the human genome contains more than 50,000 genes \cite{n-genes}. Hence, it is necessary to identify a property to leverage in order to conduct the research efficiently.

\subsection{Mutual exclusivity and coverage}

Most techniques developed in recent years for recognizing driver mutations leverage a statistical property observed in cancer patient data: each patient typically has a relatively small number of mutations that affect multiple pathways, thus each pathway will contain \textit{1 driver mutation on average} per sample. This concept of mutual exclusivity among driver mutations within the same pathway, as statistically observed in patient samples, is then axiomatized and employed by research algorithms designed to identify driver mutations \cite{multi-dendrix}. Additionally, mutual exclusivity \textit{does not affect different pathways}; it is a phenomenon that occurs exclusively within a single pathway. While the precise explanation for this occurrence is not yet fully understood, several hypotheses appear promising \cite{survey, mutual_exclusivity_expls, dendrix}:

\begin{itemize}
    \item one hypothesis is that mutually exclusive genes are functionally connected within a common pathway, acting on the same downstream effectors and creating functional redundancy; consequently, they would share the same selective advantage, meaning that the alteration of one mutually exclusive gene would be sufficient to disrupt their shared pathway, thereby removing the selective pressure to alter the others; this explanation, however, does not fully account for the phenomenon because the co-alteration of mutually exclusive genes should not result in negative effects on the cell.
    \item an alternative explanation is that the co-occurrence of mutually exclusive alterations is detrimental to cancer survival, leading to the elimination of cells that harbor such co-occurrences; moreover, some pairs of mutually exclusive genes could be \textit{synthetic lethal}, meaning that while the alteration of one gene may be compatible with cell survival, the simultaneous aberration of both genes would be lethal to the cell \todo{add example from survey paper?; also, use \href{https://www.nature.com/articles/s41467-020-20820-x}{example}? (mail "Risposte (parziali) alle questioni, ERG e SPOP")}.
\end{itemize}

In addition, another key property of driver pathways is \textbf{coverage}, i.e. driver genes constituting a driver pathway are frequently mutated across many samples.

Thus, \textit{a driver pathway consists of genes that are mutated in numerous patients, with mutations being approximately mutually exclusive}. It is also observed that pathways exhibiting these characteristics are generally shorter and comprised of fewer genes on average \cite{multi-dendrix}.

\subsection{\textit{De novo} and \textit{knowledge-based} approaches}

Although the true explanation for mutual exclusivity remains unknown, and its therapeutic potential is still uncertain, this phenomenon is frequently observed in data and is thought to potentially lead to discoveries in cancer treatment. Existing approaches can be categorized into two types: \textit{de novo} approaches, which identify mutually exclusive patterns using only genomic data from patients, and \textit{knowledge-based} methods, which integrate the analysis with external \textit{a priori} information \cite{survey}. De novo approaches might lack sufficient information as they do not utilize existing databases. Conversely, given that our understanding of gene and protein interactions in humans is still incomplete and many pathway databases fail to accurately represent the specific pathways and interactions present in cancer cells, \textit{knowledge-based} approaches may be limited by their dependence on existing data sources. Consequently, \textit{de novo} methods might yield new but potentially less accurate results, while \textit{knowledge-based} approaches may limit the discovery of novel biological insights \cite{multi-dendrix}.

\section{Mutual exclusivity formalization}

\subsection{Hard and soft mutual exclusivity}

In the statistical literature, two types of mutual exclusivity are defined: \textbf{hard} and \textbf{soft}. Hard mutual exclusivity describes events that are presumed to be strictly mutually exclusive, with the null hypothesis being that any observed overlap is due to random errors. However, in this context it is not feasible to test for hard mutual exclusivity, as this is a property observed statistically from patient data. Therefore, it is necessary to relax the constraint to soft mutual exclusivity, where two otherwise independent events overlap less than expected by chance due to some statistical interaction \cite{mutex}.

\subsection{Mutual exclusivity of a group}

For a pair of genes, soft mutual exclusivity can be assessed using the \href{https://en.wikipedia.org/wiki/Fisher\%27s_exact_test}{Fisher's exact test}. However, there is no agreed-upon method for analytically testing mutual exclusivity among more than two genes. One approach could involve checking whether each pair of genes within the group exhibits mutual exclusivity; this method, however, may be overly strict, as a gene set can exhibit a strong mutual exclusivity pattern as a whole even if no individual pairs show any \cite{mutex}.

\cleardoublepage
