\chapter{Classifying mutations} \label{chap:classifying_mutations}

placeholder \todo{introduction to the chapter}

\section{Mutations and pathways}

\subsection{Passenger and driver mutations}

In the previous sections, it was discussed how mutations play a critical role in cancer development, but not every aberration that occurs in a tumor is relevant to its proliferation. In fact, there are two types of mutations in cancer: \textbf{passenger mutations} and \textbf{driver mutations}. Passenger mutations do not confer direct benefits to tumor growth or development, whereas driver mutations actively contribute to cancer progression by providing an evolutionary advantage and promoting the proliferation of tumor cells. \textbf{Driver genes} are genes that harbor at least one driver mutation, though it may also contain passenger ones \cite{vogelstein2013}.

Some studies do not fully agree with this dichotomous model \cite{three_mutations}, the general consensus is that mutations are divided into these two categories. But further complicating matters, the term “driver gene” has two distinct meanings in cancer research. Originally, the \textit{driver-versus-passenger} concept was used to differentiate mutations that provide a selective growth advantage from those that do not. According to this definition, genes without driver mutations cannot be classified as driver genes. However, many genes that have few or no driver mutations are still referred to as driver genes in the literature. This includes genes that are overexpressed, underexpressed, or epigenetically altered in tumors. Although some of these genes might contribute significantly to cancer development, classifying all of them as \textit{driver genes} is misleading \cite{vogelstein2013}.

placeholder.\todo{expand on this telling WHY it is important to classify mutations into these two categories}

% A \textbf{driver gene} is a gene that harbors at least one driver mutation, though it may also contain passenger mutations \todo{\href{https://www.aiom.it/wp-content/uploads/2019/02/20190524RM_21_Tommasi.pdf}{DO I ADD THIS AS A CITATION???}}. A driver pathway consists of at least one driver gene. Driver mutations, genes, and pathways are of significant scientific interest due to their crucial role in cancer proliferation. \todo{expand this section}

% \subsection{Importance of pathways}
%
% placeholder \todo{questa sezione fa riscritta perché la roba che sta scritta qui non mi piace}
%
% \textbf{Cell signaling} is the process by which cells interact with each other, themselves, or their environment. It concerns the transduction of signals, which can be chemical, or can involve various types such as pressure, temperature, or light signals \cite{cell_signaling}. \textbf{Pathways} are sequences of molecular interactions within a cell that lead to a change in the cell or the production of a specific product \cite{pathway}. These pathways have a direction in which the actions occur, with the terms \textit{upstream} and \textit{downstream} indicating the initial and final stages of these processes, respectively.
%
% In cancer research, \textbf{signaling pathways} are of particular interest because they mediate the transduction of cell signals. Identifying and targeting the signaling pathways responsible for cancer growth could potentially halt the development of the disease. \todo{\href{https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8002322/}{check this out}, also check if what i wrote is actually true, i think i read it somewhere but can't find the source right now; expand on cell signaling? expand of pathways? if yes, make subsections}

\subsection{Problems with frequency analyses}

To classify mutations into the two categories described, assessing their biological function is essential, though this remains a challenging task. Numerous methods exist to predict the functional impact of mutations based on \textit{a priori} knowledge. However, these approaches often fail to integrate information effectively across various mutation types and are limited by their reliance on known proteins, rendering them less effective for less-studied ones \cite{multi-dendrix}.

With the decreasing cost of DNA sequencing, it is now possible to categorize mutations by examining their frequency, as driver mutations are typically the most recurrent in patients' genomes. For instance, some methods focus on comparing the mutation frequency of an individual gene to that of others within the same or related tumors, while accounting for sequence context and gene size. For genes with a very high number of mutations, such as TP53 or KRAS \todo{add links}, most statistical methods will strongly suggest that these genes are drivers; these highly mutated genes are often referred to as \textit{mountains} \cite{multi-dendrix, vogelstein2013}.

However, genes with more than one, but still relatively few mutations, termed \textit{hills}, are more common in cancer genome landscapes. In these cases, mutation frequency and context alone are insufficient to reliably identify driver genes, as background mutation rates can vary significantly among different patients and regions of the genome. Recent research on normal cells has shown that the mutation rate can vary more than 100-fold within the genome. In tumor cells, this variation can be even greater, affecting entire genomic regions in a seemingly random manner \cite{vogelstein2013}.

Moreover, in many cases, because driver mutations are primarily found in genes involved in cell signaling pathways, different patients may harbor mutations at different loci within the same pathway. Consequently, driver mutations can vary widely between patient samples, even within the same cancer type, resulting in minimal overlap of mutated genes across sample pairs, even from the same patient, which further reduces the statistical power of frequency-based analyses \cite{multi-dendrix, mpdfinder}.

Therefore, methods based solely on mutation frequency can only prioritize genes for expanded investigation and cannot definitively identify driver genes that are mutated at relatively low frequencies \cite{vogelstein2013}.

\subsection{Focus on pathways} \todo{continua il lavoro qua}

All \textit{known} driver genes can be categorized into one or more of \textbf{12 pathways}. These pathways can be further grouped into three fundamental cellular processes, namely \textbf{cell fate}, \textbf{cell survival} and \textbf{genome maintenance} \cite{vogelstein2013}.

Understanding these pathways is essential for comprehending patient heterogeneity. For instance, different lung cancers may harbor \textit{distinct mutations} affecting the \textbf{same cellular signaling pathways}. For instance, mutations might occur in growth factor receptors, KRAS, NF1, or BRAF, all of which impact signaling mechanisms that drive tumor growth. 

inactivation of any tumor suppressor gene is expected to lead to the activation of growth-promoting signals further downstream in the pathway.

%%%%%%%%% TODO USA LA ROBA COMMENTATA QUA

% Moreover, multiple alternative driver alterations in different genes may lead to similar downstream effects. In such instances, the selective advantage is distributed among the alterations frequencies of these genes \cite{mutex}.
%
% Therefore, studies should be conducted at the pathway level, as it is well established that different mutations can affect the same pathway across multiple samples \cite{multi-dendrix}. However, since each pathway involves multiple genes, numerous possible combinations of driver mutations could impact a crucial cancer pathway, making it computationally unfeasible to test every possible gene permutation \cite{dendrix} --- estimates suggest that the human genome contains more than 50,000 genes \cite{n-genes}. Hence, it is necessary to identify a property to leverage to search efficiently.

\subsection{Mutual exclusivity and coverage}

Most techniques developed in recent years for recognizing driver mutations leverage a statistical property observed in cancer patient data: each patient typically has a relatively small number of mutations that affect multiple pathways, thus each pathway will contain \textit{1 driver mutation on average} per sample. This concept of mutual exclusivity among driver mutations within the same pathway, as statistically observed in patient samples, is then axiomatized and employed by research algorithms designed to identify driver mutations \cite{multi-dendrix}. Additionally, mutual exclusivity \textit{does not affect different pathways}; it is a phenomenon that occurs exclusively within a single pathway. While the precise explanation for this occurrence is not yet fully understood, several hypotheses appear plausible \cite{survey, mutual_exclusivity_expls, dendrix}:

\begin{itemize}
    \item one hypothesis is that mutually exclusive genes are functionally connected within a common pathway, acting on the same downstream effectors and creating functional redundancy; consequently, they would share the same selective advantage, meaning that the alteration of one mutually exclusive gene would be sufficient to disrupt their shared pathway, thereby removing the selective pressure to alter the others; this explanation, however, does not fully account for the phenomenon because the co-alteration of mutually exclusive genes should not result in negative effects on the cell.
    \item an alternative explanation is that the co-occurrence of mutually exclusive alterations is detrimental to cancer survival, leading to the elimination of cells that harbor such co-occurrences; moreover, some pairs of mutually exclusive genes could be \textit{synthetic lethal}, meaning that while the alteration of one gene may be compatible with cell survival, the simultaneous aberration of both genes would be lethal to the cell.
\end{itemize}

An example of the latter is provided by the gene pair \href{https://en.wikipedia.org/wiki/ERG_(gene)}{ERG} and \href{https://en.wikipedia.org/wiki/SPOP}{SPOP}, which are commonly overexpressed in patients with prostate cancer, but they are mutually exclusive due to their \textit{synthetic lethality}. Wild-type (WT) SPOP facilitates the degradation of various proteins, including \href{https://en.wikipedia.org/wiki/ZMYND11Jjla}{ZMYND11}, which regulates androgen receptor (AR) signaling. Tumors with mutant ERG require reduced AR signaling to sustain their cancerous effects; therefore, mutant ERG upregulates WT SPOP to enhance the degradation of ZMYND11 and lower AR signaling. In contrast, when SPOP is mutated it loses the ability to degrade ZMYND11, leading to its accumulation and increased AR signaling. This amplified AR signaling is incompatible with the function of mutant ERG, which relies on low AR signaling. Consequently, while ERG and SPOP mutations can each support oncogenic activity individually, their simultaneous aberration is not viable due to the conflicting requirements for AR signaling \cite{erg_spop}.

In addition, another key property of driver pathways is \textbf{coverage}, i.e. driver genes constituting a driver pathway are frequently mutated across many samples. \todo{sai che cosa aggiungere, aggiungilo appena puoi}

Thus, \textit{a driver pathway consists of genes that are mutated in numerous patients, with mutations being approximately mutually exclusive}. It is also observed that pathways exhibiting these characteristics are generally shorter and comprised of fewer genes on average \cite{multi-dendrix}.

\section{Mutual exclusivity formalization}

\subsection{Challenges in quantifying mutual exclusivity}

Finding an effective method to appropriately quantify the level of mutual exclusivity is not straightforward. In the statistical literature, two types of mutual exclusivity are defined: \textit{hard} and \textit{soft}. Hard mutual exclusivity describes events that are presumed to be strictly mutually exclusive, with the null hypothesis being that any observed overlap is due to random errors. However, in this context, it is not feasible to test for hard mutual exclusivity, as this is a property observed statistically from patient data. Therefore, it is necessary to relax the constraint to soft mutual exclusivity, where two otherwise independent events overlap less than expected by chance due to some statistical interaction.

Moreover, while soft mutual exclusivity of a pair of genes can be assessed using the \href{https://en.wikipedia.org/wiki/Fisher\%27s_exact_test}{Fisher's exact test}, there is no agreed-upon method for analytically testing mutual exclusivity among more than two genes. For instance, one intuitive approach could involve checking whether each pair of genes within the group exhibits mutual exclusivity; this method, however, may be overly strict, as a gene set can exhibit a strong mutual exclusivity pattern as a whole even if no individual pairs show any \cite{mutex}.

Due to the complexity of measuring mutual exclusivity, recent papers have proposed various formulations, based on different assumptions, which will be discussed in later sections.

Finding the solution to this problem is computationally difficult even for small values of $k$ (e.g. there are $\approx 10^{23}$ subsets of size $k = 6$ of 20,000 genes), and it can be proven that it is NP-Hard. \todo{scrivi quello che sai che devi scrivere}

\subsection{A deterministic formalization of mutual exclusivity}

\textcite{dendrix}, the authors of a landmark paper that developed an algorithm called Dendrix, provided the following mathematical formalization for the properties of mutual exclusivity and coverage for a set of genes.

\begin{definition}[Mutation matrix] \label{mut_matrix_def}
    A \textbf{mutation matrix} is a matrix with $m$ rows and $n$ columns, where each row represents a patient and each column represents a gene, and the entry $a_{i, j}$ is equal to 1 if and only if gene $j$ is mutated in patient $i$.
\end{definition}

\begin{example}[Mutation matrix] \label{mutation_matrix}
    An example of a mutation matrix is the following:

    \begin{table}[H]
        \centering
        \begin{tabular}{c|ccc}
                  & $g_1$ & $g_2$ & $g_3$ \\
            \hline
            $p_1$ & 0 & 1 & 0 \\
            $p_2$ & 1 & 1 & 0 \\
            $p_3$ & 0 & 0 & 1 \\
        \end{tabular}
        \caption{A mutation matrix.}
    \end{table}
\end{example}

\begin{definition}[Coverage of a gene]
    Given a gene $g$, the \textbf{coverage of $g$} denotes the set of patients which have $g$ mutated, and it is defined as follows $$\Gamma(g) := \{i \mid a_{i, g} = 1\}$$ 
\end{definition}

Under the previous definitions of mutual exclusivity, $M$ is \textbf{mutually exclusive} if no patient has more than one mutated gene, formally $$\forall g, g' \in M \quad \Gamma(g) \cap \Gamma(g') = \varnothing$$

\begin{definition}[Coverage of a set]
    Given a set $M$ of genes, the \textbf{coverage of $M$} denotes the set of patients in which at least one of the genes in $M$ is mutated, and it is defined as follows $$\Gamma(M) := \bigcup_{g \in M}{\Gamma(g)}$$
\end{definition}

Any gene set can be thought of as a $m \times k$ submatrix of a mutation matrix $A$, up to rearranging $A$'s columns --- their order does not matter since they represent genes. Accordingly, such a submatrix is said to be \textbf{mutually exclusive} if each row contains at most one 1.

Furhermore, given a gene set $M$, the following properties are formalized:

\begin{enumerate}[label=\roman*), font=\itshape]
    \item \textit{coverage}: most patients have at least one mutation in $M$;
    \item \textit{approximate exclusivity}: most patients have exactly one mutation in $M$.
\end{enumerate}

To evaluate these two attributes, a measure that quantifies the trade-off between coverage and mutual exclusivity is introduced.

\begin{definition}[Coverage overlap] \label{cov_over}
    Given a set $M$ of genes, the \textbf{coverage overlap} is defined as follows: $$\omega(M) := \sum_{g \in M}{\abs{\Gamma(g)}} - \abs{\Gamma(M)}$$
\end{definition}

Note that the sum in \cref{cov_over} is the number of 1s in $M$'s corresponding submatrix.

\begin{example}[Coverage overlap]
    Considering the mutation matrix in \cref{mutation_matrix}, if $M=\{g_1, g_2\}$ then $$\omega(M)=\abs{\Gamma(g_1)} + \abs{\Gamma(g_2)} - \abs{\Gamma(\{g_1, g_2\})} = \abs{\{p_2\}} + \abs{\{p_1, p_2\}} - \abs{\{p_1, p_2\}} = 1 + 2 - 2 = 1$$
\end{example}

Indeed, $\omega(M)$ is the \textit{number of patients that are counted more than once in the sum}. Note that $\omega(M) \ge 0$, with equality holding only if the sum equals the coverage of $M$, which means that no patient has more than one mutated gene of $M$.

\begin{definition}[Mutually exclusive set]
    A gene set $M$ is considered to be \textbf{mutually exclusive} if $\omega(M) = 0$.
\end{definition}

\begin{definition}[Weight of gene set] \label{weight}
    Given a set of genes $M$, to take into account both coverage and coverage overlap, the following measure is introduced: $$W(M): = \abs{\Gamma(M)} - \omega(M) = 2 \abs{\Gamma(M)} - \sum_{g \in M} {\abs{\Gamma(g)}}$$
\end{definition}

Note that $W(M) = \Gamma(M)$ when $M$ is mutually exclusive.

In order to find an optimal gene set, the following problem has to be solved:

\begin{displayquote}\label{mwsp}
    \textbf{Maximum Weight Submatrix Problem} (MWSP): Given an $m \times n$ mutation matrix $A$, and an integer $k > 0$, find a $m \times k$ submatrix of $A$ that maximizes $W(M)$.
\end{displayquote}

The following proof shows that this problem is NP-Hard.

\begin{theorem}[NP-Hardness of the MWSP]
    The Maximum Weight Submatrix Problem is NP-Hard.
\end{theorem}

\begin{proof}
        The proof is by reduction from the \href{https://en.wikipedia.org/wiki/Independent_set_(graph_theory)}{Independent Set} Problem (ISP), which is known to be NP-Hard \cite{hochbaum}. In the ISP, it is asked whether there is an independent set of size $k$ in a given graph $G$. An independent set for $G = (V, E)$ is a set of vertices $I \subseteq V(G)$ such that there is no edge among the vertices of $I$, i.e. $$\forall u, v \in I \mid u \neq v \quad (u, v) \notin E(G)$$

        Given an instance of the ISP, a mutation matrix representing an instance of the MWSP is built in polynomial time as follows:

        \begin{itemize}
            \item let $\Delta := \max_{v \in G}{\deg(v)}$, and for each $v \in V(G)$ let $\mathcal S_v := \left\{s_v^{(1)}, \ldots, s_v^{(\Delta - \deg(v))}\right\}$ be a set of variables; also, consider the following set $$\mathcal S := \left\{s_e \mid e \in E(G)\right\} \cup \rbk{\bigcup_{v \in V(G)}{\mathcal S_v}}$$
            \item build a matrix $A$ of size $\abs{\mathcal S} \times \abs{V(G)}$, as illustrated below

                \begin{table}[H]
                    \centering
                    \begin{tabular}{c|ccc}
                              & $v_1$ & $\ldots$ & $v_n$ \\
                        \hline
                        $e_1$ & $\ddots$ & & \\
                        $\vdots$ & & $\ddots$ & \\
                        $e_m$ & & & $\ddots$ \\
                        \hline
                        $s_{v_1}^{(1)}$ & $\ddots$ & & \\
                        $\vdots$ & & $\vdots$ & \\
                        $s_{v_1}^{(\Delta - \deg(v_1))}$ & &  & \\
                        $\vdots$ & & $\vdots$ & \\
                        $s_{v_n}^{(1)}$ & &  & \\
                        $\vdots$ & & $\vdots$ & \\
                        $s_{v_n}^{(\Delta - \deg(v_n))}$ & & & $\ddots$
                    \end{tabular}
                    \caption{The described matrix.}
                \end{table}

            \item define $A$'s cells as follows: $$a_{s, v} = 1 \iff s = s_{(u, v)}, u \in V(G) \lor s \in \mathcal S_v$$ which means that $a_{s, v}$ will be 1 if and only if $s$ is either a variable from the set $\{s_e \mid e \in E(G)\}$ where the edge $e$ has $v$ as endpoint, or $s$ is a variable defined in $\mathcal S_v$.
        \end{itemize}

        Note that:

        \begin{enumerate}[label=\roman*), font=\itshape]
            \item $\forall v \in V(G) \quad \abs{\Gamma(v)} = \Delta$ due to the added variables at the end of each column;
            \item $\forall u, v \in V(G) \quad \Gamma(u) \cap \Gamma(v) \neq \varnothing \iff (u, v) \in E$, since no pair of columns can have a 1 in the same row in the second half of $A$ by definition of the sets $\mathcal S_{v_1}, \ldots , \mathcal S_{v_n}$, therefore $\Gamma(u)$ and $\Gamma(v)$ can have an intersection if and only if there is an edge $(u, v) \in E(G)$.
        \end{enumerate}

        Hence, consider a set $M = \{v_1, \ldots, v_k\}$ of $k$ columns of $A$. Note that:

        \begin{itemize}
            \item from $(i)$ it follows that $$\sum_{i = 1}^k {\abs{\Gamma \rbk{v_i}}} = k \Delta$$ and consequently $\abs{\Gamma(M)} \le k \Delta$, meaning that the largest value $\abs{\Gamma(M)}$ can have is $k \Delta$; hence, from the equation in \cref{weight}, it follows that the maximum value $W(M)$ can reach is
                \begin{equation*}
                    \begin{split}
                        W(M) & = 2 \abs{\Gamma(M)} - \sum_{i = 1}^k{\abs{\Gamma \rbk{v_i}}} \\
                             & = 2 k \Delta - k \Delta \\
                             & = k \Delta
                    \end{split}
                \end{equation*}

            \item from $(ii)$ it follows that $\abs{\Gamma(M)} = k \Delta \iff \forall u, v \in V(G) \quad \Gamma(u) \cap \Gamma(v) = \varnothing \iff \forall u, v \in V(G) \quad (u, v) \notin E(G) \iff M$ is an independent set, by definition.
        \end{itemize}

        This means that $W(M)$ is maximized if and only if $M$ is an independent set; therefore, the MWSP can be solved on $A$ if and only if the ISP can be solved on $G$.
\end{proof}

\subsection{Extending the deterministic equation} \label{multi_dendrix_2nd_chap} \todo{considera di spostare questo alla fine}

\textcite{multi-dendrix} refine the weight function of \textcite{dendrix}, aiming to extend the metric to assess mutual exclusivity across multiple driver pathways. In particular, while identifying individual driver pathways is crucial, most cancer patients are likely to have driver mutations across multiple pathways.

To effectively identify multiple driver pathways, it is necessary to establish criteria for evaluating potential \textit{collections of gene sets}. Based on the same biological reasoning mentioned earlier, it is expected that each pathway will contain approximately one driver mutation. Furthermore, since each driver pathway is crucial for cancer development, it is expected that most patients will harbor a driver mutation in most driver pathways. Consequently, high exclusivity is predicted within the genes of each pathway, along with high coverage of each pathway individually. One metric that meets these criteria is to find a collection $M = \{M_1, \ldots, M_t\}$ of gene sets which maximizes the sum of individual weights, i.e. $\sum_{\rho = 1}^t {W(M_\rho)}$.

\subsection{A statistical approach}

\textcite{mutex} criticize the metric developed by \textcite{dendrix} because it has a strong bias toward highly mutated genes, and in some instances, the excessive emphasis on coverage leads to false positives and negatives. \todo{ci sono esempi in file supplementari, li guardo?}. They propose a metric that extends Fisher's exact test --- also known as \textit{hypergeometric test} --- to quantify the mutual exclusivity between multiple measurements.

Specifically, the alteration of a pair of genes is defined to be \textbf{mutually exclusive} \textit{if their overlap in samples is significantly less than expected by chance}, and this can be assessed through a \textit{hypergeometric test}. It is important to note that a uniform alteration frequency across may not always hold, particularly for hyper-mutated samples often resulting from prior mutations in DNA repair mechanisms. Addressing this heterogeneity is challenging, as each overlap in the null model has a different probability. This remains an open problem, and to partially mitigate it, albeit at the cost of statistical power, hyper-altered samples are excluded from the analysis.

\textcite{mutex} also developed a metric to assess the mutual exclusivity of a group of genes. Consider the following null hypothesis:

\begin{displayquote}
    $H_0$: \textit{The specific member gene in the group is altered independently from the union of other alterations in the group}.
\end{displayquote}

Using Dendrix's notation, $H_0$ states that for a given gene set $M$, for every gene $j \in M$, mutations in $\Gamma(j)$ are independent of alterations in $\Gamma(M - \{j\})$. In brief, $H_0$ states that any observed pattern of mutual exclusivity among gene alterations is due to random chance, not due to any underlying biological or oncogenic mechanism.

$H_0$ is then tested for each $j \in M$ by evaluating the co-distribution of $i$ with the union of the others through Fisher's exact test, generating $\abs{M}$ $p$-values. These $p$-values represent the probabilities for the independent distribution of each member gene. To ensure that every group member contributes to the pattern, the least significant --- i.e. the largest --- $p$-value of the group is used as the initial score of the group. With Dendrix's notation

\begin{equation}
    s_0 := \max_{j \in M} {H \abk{\Gamma(j), \Gamma(M - \{j\})}}
\end{equation}

where $s_0$ is the initial score, and $H$ is the hypergeometric test.

Since multiple groups are being tested, $s_0$ is affected by \href{https://en.wikipedia.org/wiki/Multiple_comparisons_problem}{\textit{multiple hypothesis testing}}, because the probability of finding statistically significant results may increase by chance. To account for it, \todo{finish this section}



From this second set of $p$-values, the least significant one is selected as the multiple hypothesis testing corrected final score. \todo{talk about the last paragraph, which is even less comprehensible}

\subsection{A clustering approach} \label{c3_chap2} \todo{change this title soo similar with the previous one}

Another notable approach utilized in several papers involves constructing gene graphs and identifying clusters based on specific criteria; this method is demonstrated by \textcite{c3}.

Let $G = (V, E)$ be a \textit{complete graph} of genes, thus an edge exists between any pair of vertices. Each edge $(u, v) \in E(G)$ is assigned two weights:

\begin{itemize}
    \item a \textbf{positive weight} $w_{uv}^+$, which represents \textit{the cost of placing $u$ and $v$ in different clusters};
    \item a \textbf{negative weight} $w_{uv}^-$, which represents \textit{the cost of placing $u$ and $v$ in the same cluster};
\end{itemize}

i.e. by making $w_{uv}^+$ large, placing $u$ and $v$ in different clusters is discouraged, and viceversa; the same concept applies for $w_{uv}^-$. Indeed, as weights representations suggest, \textit{genes in the same cluster are likely to be mutually exclusive}.

Weights are calculated using four types of datasets: gene mutation data, \href{https://www.genome.gov/genetics-glossary/Copy-Number-Variation}{copy number variation (CNV)}, network information, and gene expression data. To appropriately combine the sources from which the information was obtained, linear combinations were utilized to account for the reliability of the sources from which the data was drawn.

Additionally, as each type of data contributes differently to the driver discovery process, linear combinations are used, based on the importance or accuracy of each, specifically:

\begin{itemize}
    \item the $(\mathrm e)$ label refers to \textit{exclusivity};
    \item the $(\mathrm c)$ label refers to \textit{coverage};
    \item the $(\mathrm n)$ label refers to \textit{network information};
    \item the $(\mathrm x)$ label refers to \textit{expression data}.
\end{itemize}

Let $A$ be an $m \times n$ mutation matrix, as described in \cref{mut_matrix_def}. In addition, let $C$ be an $m \times n$ matrix representing the CNV data, where $c_{i, j} = 0$ means that there is no change in the copy number of gene $j$ in sample $i$, otherwise, the corresponding number reflects the deviation of the CNV number from its baseline --- hence, $C$ contains both positive and negative values. Following this, a binary matrix $M$ is constructed combining $A$ and $C$ as follows:

\begin{equation}
    m_{i, j} = 0 \iff \soe{l}{a_{i, j} = 0 \\ l_{\mathrm{cnv}} < c_{i, j} < h_{\mathrm{cnv}}}
\end{equation}

where $l_{\mathrm{cnv}}$ and $h_{\mathrm{cnv}}$ are lower and upper bounds on copy numbers that determine the significance level. \todo{menziono i valori che hanno usato loro? menziono le conseguenze che valori alti/bassi di queste soglie hanno, secondo loro?} Therefore, if $m_{i, j} = 0$, no mutation of gene $j$ is recorded in sample $i$, otherwise gene $j$ is \textit{deemed mutated}.

\begin{definition}[Coverage of a vertex]
    Given a vertex $u \in V(G)$, i.e. a gene, the \textbf{coverage of $u$} $$\mathscr{S}(u) := \{i \mid m_{i, u} = 1\}$$ denotes the set of patients in which $u$ is altered.
\end{definition}

Note that $\mathscr{S}(u)$ corresponds to $\Gamma(u)$ under Dendrix's notation, but is defined through the $M$ matrix respectively.

\begin{definition}[Mutual exclusivity component] \label{me_comp}
    The \textbf{mutual exclusivity component} between two genes $u, v \in V(G)$ is defined as follows: $$w_{uv}^-(\mathrm e) := a \cdot \dfrac{\abs{\mathscr{S}(u) \cap \mathscr{S}(v)}}{\min(\abs{\mathscr{S}(u)}, \abs{\mathscr{S}(v)})}$$ where $a$ is a user-defined scaling parameter.
\end{definition}

This ratio is often referred to as \textbf{IoM} (\textit{Intersection over Minimum}), and suits the criteria of mutual exclusivity because the fewer patients who have both $u$ and $v$ mutated, the smaller the weight, making it more plausible that $u$ and $v$ are mutually exclusive, therefore the cost of placing them in the same cluster should be low. Note that

\begin{equation}\label{neg_weight_constraint}
    \forall u, v \in V(G) \quad a = 1 \implies 0 \le w_{uv}^-(\mathrm e) \le 1
\end{equation}

\begin{definition}[Negative weights] \label{neg_weights}
    \textbf{Negative weights} only depend on the mutual exclusivity component, i.e. $$\forall u, v \in V(G) \quad w_{uv}^- := w_{uv}^-(\mathrm e)$$
\end{definition}

By contrast, positive weights can depend on multiple factors which will be presented in the following sections \todo{va bene se la metto su questo piano?}. Focusing on \textbf{coverage}, if two genes $u$ and $v$ increase the coverage of the set significantly, $w_{uv}^+(\mathrm c)$ should be large such that they are encouraged to be placed in the same cluster. Let

\begin{equation}
    D(u, v) := \abs{\mathscr{S}(u) \Delta \mathscr{S}(v)}
\end{equation}

where $\Delta$ denotes the symmetric difference of two sets; a large value of $D(u, v)$ suggests that $u$ and $v$ should be placed in the same cluster. Also, let

\begin{equation}
    \mathscr{D} := \{D(u, v) \mid u, v \in V(G)\}
\end{equation}

and let $T(J)$ be the $J$-th percentile of the values in $\mathscr{D}$.

\begin{definition}[Coverage component] \label{co_comp}
    The \textbf{coverage component} is defined as follows: $$w_{uv}^+(\mathrm c) := \soe{ll}{1 & D(u, v) > T(J) \\ \dfrac{D(u, v)}{T(J)} & D(u, v) \le T(J)}$$
\end{definition}

Note that, similar to \cref{neg_weight_constraint}

\begin{equation}
    \forall u, v \in V(G) \quad 0 \le w_{uv}^+(\mathrm c) \le 1
\end{equation}

The linear combinations that define $w_{uv}^+$ will be discussed afterward.

\cleardoublepage
